network/src/legacy/mod.rs

```diff
@@ -755,19 +755,6 @@ impl PolkadotProtocol {
 					warn!(target: "polkadot_network", "Encountered tracked but disconnected validator {:?}", primary),
 			}
 		}
-
-		let availability_store = self.availability_store.clone();
-		let collation_cloned = collation.clone();
-
-		async move {
```

Comment:
So, this doesn't need to happen here?

Reply:
IMO no, the Collator doesn't need to keep it available, at least not via this API. it's something to revisit in a later refactoring.